cpp/src/Ice/Exception.cpp

```diff
@@ -116,40 +85,40 @@ namespace
         assert(pc == 0);
         return 0;
     }
-
+#elif defined(ICE_BACKTRACE)
+    bool backTraceEnabled = false;
 #endif
 
+#ifdef ICE_DBGHELP
     class Init
     {
     public:
-        Init()
-        {
-#ifdef ICE_LIBBACKTRACE
-            // Leaked, as libbacktrace does not provide an API to free this state.
-            //
-            bstate = backtrace_create_state(0, 1, ignoreErrorCallback, 0);
-
-            // The first call to backtrace_pcinfo does not initialize bstate->fileline_fn
-            // in a thread-safe manner, see https://gcc.gnu.org/bugzilla/show_bug.cgi?id=81098.
-            // So we make a "dummy" call to backtrace_pcinfo to initialize it here.
-            //
-            backtrace_pcinfo(bstate, 0, ignoreFrame, ignoreErrorCallback, 0);
-#endif
-        }
-
         ~Init()
         {
-#ifdef ICE_DBGHELP
-            if (process != 0)
+            if (process)
             {
                 SymCleanup(process);
-                process = 0;
+                CloseHandle(process);
+                process = nullptr;
             }
-#endif
         }
     };
 
     Init init;
+#endif
+
+    inline bool collectStackTraces() noexcept
```

Comment:
Should we be using a memory barrier to access these global variables. Isn't there a chance they'll be stale otherwise.

Reply:
Added mutex lock.